assets/app/scripts/controllers/overview.js

```diff
@@ -384,7 +417,98 @@ angular.module('openshiftConsole')
       });
     });
 
+    /*
+     * The topology graph wants unique identifiers.
+     *
+     * One of the warts on the kubernetes/openshift API is that there
+     * is no unique global identifier per item. We construct them here from
+     * Kind/metadata.name ... which is good enough in a single namespace.
+     */
+    function makeId(name, kind) {
+      return kind + "/" + name;
+    }
+
+    var updateTimeout = null;
+
+    function updateTopology() {
+      updateTimeout = null;
+
+      var topologyRelations = [];
+      var topologyItems = { };
+
+      /*
+       * Put each of the items into the topology. Each of these
+       * are name -> object mappings.
+       */
+      [ $scope.pods,
+        $scope.routes,
+        $scope.services,
+        $scope.deployments,
+        $scope.deploymentConfigs
+      ].forEach(function(items) {
+        angular.forEach(items, function(item, name) {
+          var id = makeId(name, item.kind);
+          topologyItems[id] = item;
+        });
+      });
+
+      /*
+       * Most of the tables in this scope are in a standard form with
+       * string keys, pointing to a map of further name -> object mappings.
+       *
+       * We can process them all together.
+       */
+      [
+        [ $scope.podsByService, "Service", "Pod" ],
+        [ $scope.podsByDeployment, "ReplicationController", "Pod" ],
+        [ $scope.deploymentsByService, "Service", "ReplicationController" ],
+        [ $scope.deploymentConfigsByService, "Service", "DeploymentConfig" ],
+        [ $scope.routesByService, "Service", "Route" ]
+      ].forEach(function(args) {
+        var sourceKind = args[1];
+        var targetKind = args[2];
+        angular.forEach(args[0], function(items, name) {
+          var sourceId;
+          if (name) {
+            sourceId = makeId(name, sourceKind);
+            angular.forEach(items, function(item, name) {
+              topologyRelations.push({ source: sourceId, target: makeId(name, targetKind) });
+            });
+          }
+        });
+      });
+
+      $scope.topologyItems = topologyItems;
+      $scope.topologyRelations = topologyRelations;
+      $scope.$digest();
+    }
+
+    function updateTopologyLater() {
+      if (!updateTimeout)
+        updateTimeout = window.setTimeout(updateTopology, 100);
+    }
+
+    $scope.$on("select", function(ev, item) {
+      $scope.topologySelection = item;
+      if (item)
+        ObjectDescriber.setObject(item, item.kind);
+      else
+        ObjectDescriber.clearObject();
+      $scope.$digest();
```

Comment:
```$scope.$apply()``` always digests the ```$rootScope``` (and thus all the scopes in the app). Not necessary in this case. However this is your call. I'll go ahead and make the changes.